platform/linuxbsd/freedesktop_portal_desktop.cpp

```diff
@@ -617,6 +749,262 @@ void FreeDesktopPortalDesktop::_system_theme_changed_callback() {
+				success = _dbus_messsage_iter_append_basic_variant(&reply_iter, DBUS_TYPE_STRING, item.tooltip.utf8());
+			} else if (strcmp(property, "ItemIsMenu") == 0) {
+				success = _dbus_message_iter_append_bool_variant(&reply_iter, false);
+			} else if (strcmp(property, "Menu") == 0) {
+				success = _dbus_messsage_iter_append_basic_variant(&reply_iter, DBUS_TYPE_OBJECT_PATH, "/");
+			} else {
+				reply = dbus_message_new_error(message, DBUS_ERROR_UNKNOWN_PROPERTY, vformat("Property %s not found.", property).utf8());
+
+				// Replying with an error is intended here so... Failed successfully?
+				success = true;
+			}
+
+			if (!success) {
+				reply = dbus_message_new_error(message, DBUS_ERROR_FAILED, "Failed to setup message.");
+				ERR_PRINT("Failed to setup message.");
+			}
+
+			ERR_FAIL_NULL_V(reply, DBUS_HANDLER_RESULT_HANDLED);
+
+			dbus_uint32_t serial = dbus_message_get_serial(message);
+			dbus_connection_send(connection, reply, &serial);
+
+			dbus_message_unref(reply);
+
+			return DBUS_HANDLER_RESULT_HANDLED;
+		}
+	}
+
+	return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
+}
+
+bool FreeDesktopPortalDesktop::indicator_create(DisplayServer::IndicatorID p_id, const Ref<Image> &p_icon) {
+	MutexLock mutex_lock(file_dialog_mutex);
+
+	ERR_FAIL_COND_V(indicators.has(p_id), false);
+
+	if (unsupported) {
+		return false;
+	}
+
+	DBusError error;
+	dbus_error_init(&error);
+
+	DBusConnection *bus = dbus_bus_get(DBUS_BUS_SESSION, &error);
+	if (dbus_error_is_set(&error)) {
+		if (OS::get_singleton()->is_stdout_verbose()) {
+			ERR_PRINT(vformat("Error opening D-Bus connection: %s", error.message));
+		}
+		dbus_error_free(&error);
+		unsupported = true;
+		return false;
+	}
+
+	int pid = OS::get_singleton()->get_process_id();
+
+	String name = vformat("org.freedesktop.StatusNotifierItem-%d-%d", pid, p_id);
+	dbus_bus_request_name(bus, name.utf8(), DBUS_NAME_FLAG_DO_NOT_QUEUE, &error);
+	if (dbus_error_is_set(&error)) {
+		dbus_error_free(&error);
+		return false;
+	}
+
+	indicator_id_map[name] = p_id;
+
+	StatusNotifierItem &item = indicators[p_id];
+	item.id = p_id;
+	item.connection = bus;
+	item.name = name;
+
+	// TODO: Error handling.
+	indicator_set_icon(p_id, p_icon);
+
+	const char *name_ptr = name.utf8();
+
+	DBusMessage *message = dbus_message_new_method_call(
+			BUS_STATUS_NOTIFIER_WATCHER_NAME, BUS_STATUS_NOTIFIER_WATCHER_PATH, BUS_INTERFACE_STATUS_NOTIFIER_WATCHER,
+			"RegisterStatusNotifierItem");
+	dbus_message_append_args(
+			message,
+			DBUS_TYPE_STRING, &name_ptr,
+			DBUS_TYPE_INVALID);
+
+	dbus_connection_send(bus, message, NULL);
+	dbus_message_unref(message);
+
+	item.registered = true;
+
+	return true;
+}
+
+Error FreeDesktopPortalDesktop::indicator_set_icon(DisplayServer::IndicatorID p_id, const Ref<Image> &p_icon) {
+	MutexLock mutex_lock(file_dialog_mutex);
+
+	ERR_FAIL_COND_V(!indicators.has(p_id), ERR_UNCONFIGURED);
+	ERR_FAIL_COND_V(p_icon->get_size() == Size2i(), ERR_INVALID_PARAMETER);
```

Comment:
This won't fire if the size is `{ 0, 1 }` or `{ 1, 0 }` etc., only if both axes are zero